inductiva/resources/machine_groups.py

```diff
@@ -50,24 +50,28 @@ def get_by_name(machine_name: str):
     try:
         api = compute_api.ComputeApi(inductiva.api.get_client())
         response = api.get_vm_group_by_name({"name": machine_name}).body
-        mg_class = _get_machine_group_class(response["type"],
-                                            response["is_elastic"])
-        return mg_class.from_api_response(response)
+        mg_class, kwargs = _get_machine_group_class(response["type"],
+                                                    response["is_elastic"],
+                                                    response["provider_id"])
+        return mg_class.from_api_response(response, **kwargs)
     except inductiva.client.ApiException as api_exception:
         raise api_exception
 
 
-def _get_machine_group_class(machine_type: str, is_elastic: bool):
+def _get_machine_group_class(machine_type: str, is_elastic: bool,
+                             provider: str):
     """Returns the class of the machine group"""
+    kwargs = {}
     if is_elastic:
         mg_class = resources.ElasticMachineGroup
     elif machine_type == "standard":
         mg_class = resources.MachineGroup
+        kwargs = {"provider": provider}
```

Comment:
Why do we only use `kwargs` when `machine_type == "standard"`? Shouldn't the back end also send the `provider_id` for other machine types?

Reply:
Not part of the constructor.

Reply:
Maybe we can add the provider to the BaseMachineGroup. This would make the code more uniform and it would make sense to me that every resource has a provider. If BaseMachineGroup had a provider we wouldn't need to be using kwargs (kwargs can make the code a bit harder to understand and to follow).